core/src/main/java/com/linecorp/armeria/server/healthcheck/HealthCheckServiceBuilder.java

```diff
@@ -306,7 +331,7 @@ public HealthCheckService build() {
         checkState(startHealthy || updateHandler != null,
                    "Healthiness must be updatable by server listener or update handler.");
         return new HealthCheckService(healthCheckers.build(),
-                                      healthyResponse, unhealthyResponse,
+                                      healthyResponse, degradedResponse, unhealthyResponse,
```

Comment:
We could also add `underMaintenanceResponse`?

Reply:
Seems like the number of arguments for the responses increases(`healthyResposne`, `unhealthyResponse`, `underMaintenanceResponse`, `degradedResponse`). Should we receive the responses as `Collections` instead of  adding it as an argument? This way, we won't have to add another argument when another `HealthStatus` is added.

Reply:
Yes, we can. What do you think of the below responses for each responses?(I'm a bit worried if this will break backward compatibility though)

`healthyResponse`: `{"healthy":true, "status": "HEALTHY"}`
`unhealthyResponse`: `{"healthy":true, "status": "DEGRADED"}`
`degradedResponse`: `{"healthy":true, "status": "UNDER_MAINTENANCE"}`
`unhealthyResponse`: `{"healthy":false, "status": "UNHEALTHY"}`

Reply:
I'd prefer adding `status` for all responses.

Reply:
I've added status field to all 5 responses 
- healthyResponse 
- degradedResponse
- serverStoppingResponse 
- underMaintenanceResponse
- unhealthyMaintenance